cloud_foundry_api/datadog_checks/cloud_foundry_api/cloud_foundry_api.py

```diff
@@ -130,41 +134,173 @@ def get_oauth_token(self):
         self._token_expiration = int(time.time()) + payload["expires_in"]
         self.service_check(UAA_SERVICE_CHECK_NAME, CloudFoundryApiCheck.OK, tags=sc_tags)
 
-    def scroll_pages(self, url, params, headers, additional_tags):
-        # type: (str, Dict[str, Any], Dict[str, str], List[str]) -> Dict[str, Event]
-        page = 1
-        events = {}  # type: Event
-        scroll = True
-        sc_tags = ["api_url:{}".format(urlparse(self._api_url)[1])] + self._tags
-        while scroll:
-            params = copy.deepcopy(params)
-            params["page"] = page
-            self.log.debug("Fetching events page %s", page)
-            try:
-                res = self.http.get(url, params=params, headers=headers)
-            except RequestException:
-                self.log.exception("Error connecting to the Cloud Controller API")
-                self.service_check(API_SERVICE_CHECK_NAME, CloudFoundryApiCheck.CRITICAL, tags=sc_tags)
-                return events
+    def get_orgs(self):
+        orgs = {}
+        self.get_oauth_token()
+        if self._api_version == "v2":
+            params = {
+                "results-per-page": MAX_PAGE_SIZE_V2,
+            }
+            headers = {"Authorization": "Bearer {}".format(self._oauth_token)}
```

Comment:
It seems that this line is repeated multiple times. Maybe we can create a utility function `get_auth_headers` that would refresh the `get_oauth_token` and return this dict?